commands/import.go

```diff
@@ -184,24 +208,37 @@ func (i *Importer) run() error {
 		return err
 	}
 
+	i.res = resources.InitResources(i.configFile)
+
 	// set up target database
 	i.res.DB.SelectDB(i.targetDatabase)
 
-	// get settings from an existing database
+	// set up the rolling configuration
+	// grab the current rolling settings from the MetaDB
 	exists, isRolling, currChunk, totalChunks, err := i.res.MetaDB.GetRollingSettings(i.targetDatabase)
 	if err != nil {
 		return cli.NewExitError(fmt.Errorf("\n\t[!] Error while reading existing database settings: %v", err.Error()), -1)
 	}
-	// determine the new rolling settings based on current and supplied arguments
-	rollingCfg, err := setRolling(exists, isRolling, currChunk, totalChunks,
-		i.rolling, i.currentChunk, i.totalChunks, i.res.Config.S.Rolling.DefaultChunks)
+
+	// validate the user given flags against the rolling settings from the MetaDB
+	// and determine the rolling configuration
+	rollingCfg, err := parseFlags(
+		exists, isRolling, currChunk, totalChunks,
+		i.userRolling, i.userCurrChunk, i.userTotalChunks, i.res.Config.S.Rolling.DefaultChunks,
+		i.deleteOldData,
+	)
 	if err != nil {
 		return cli.NewExitError(err.Error(), -1)
 	}
-	// preserve the default chunks setting (even though we don't use it after this currently)
-	rollingCfg.DefaultChunks = i.res.Config.S.Rolling.DefaultChunks
 	i.res.Config.S.Rolling = rollingCfg
 
+	if i.deleteOldData {
+		err := i.handleDeleteOldData()
+		if err != nil {
+			return cli.NewExitError(fmt.Errorf("Error deleting old data: %v", err.Error()), -1)
+		}
+	}
```

Comment:
When the `--delete` flag is specified I'd like to have the database be deleted as late as possible. Right now, there are several places where errors can happen after the database is deleted that cause the user to have their database removed and no new one to replace it.

```bash
# I recreated the database rtest before running each of these

# 1) trying to import when InternalSubnets is not defined
$ ./rita import --delete --config etc/rita.yaml ignore/dnscat-tsv/conn.log rtest
	[+] Removing database: rtest
	[-] Successfully deleted database rtest.

Internal subnets are not defined. Please set the InternalSubnets section of the config file.

# 2) trying to import a directory that exists but is empty, or only contains unsupported log files
$ ./rita import --delete ignore/empty rtest
	[+] Removing database: rtest
	[-] Successfully deleted database rtest.

	[+] Importing [ignore/empty]:

	[!] No compatible log files found

# 3) similar to the last one, trying to import a log file that is unsupported
$ ./rita import --delete ignore/dnscat-tsv/weird.log rtest
	[+] Removing database: rtest
	[-] Successfully deleted database rtest.

	[+] Importing [ignore/dnscat-tsv/weird.log]:
	[-] Verifying log files have not been previously parsed into the target dataset ... 
	[!] All files in this directory have already been parsed into database:  rtest
```

Fixing 1 should be as easy as moving this block below the check for internal subnets on lines 243-245.

Fixing 2 and 3 is trickier but doable. I think if we move [these lines](https://github.com/activecm/rita/blob/508-reimport/parser/fsimporter.go#L82-L108) to their own function and call it before we delete the database/chunk it would fix those.